Reader: add image carousel

arcangelini opened the pull request:
Fixes https://linear.app/a8c/issue/CML-862/reader-gallery-not-working

## Proposed Changes

* Add smol React modal for carousel images.
* Add new carousel modal to tiled gallery and gallery blocks.

## Why are these changes being made?

Currently there is no carousel to view image galleries from the reader. On the site Jetpack adds a fancy image carousel with a lot of extra features like comments and exif data. This is just a light-weight simple version of that which is more appropriate for the Reader.

## Testing Instructions

1. Use Calypso Live link and go to `/reader/blogs/140097238/posts/169227718` or `/reader/blogs/4/posts/9442`
2. Check that all the images work and styling looks the same as the website.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

arcangelini committed:
Reader: add image carousel

davemart-in commented:
When I test this on [Calypso.live](https://container-awesome-northcutt.calypso.live/reader/blogs/4/posts/9442), I'm not seeing the buttons, or the text showing which image I'm on for some reason:

<img width="2936" height="1502" alt="CleanShot 2025-09-02 at 13 55 53@2x" src="https://github.com/user-attachments/assets/c27d8f24-dd18-4245-bb9a-9015ea66e8ed" />

arcangelini committed:
Styling match

arcangelini committed:
Final draft - moved component and styling matches

arcangelini commented:
Thanks @davemart-in just pushed the final version. I moved things around and simplified it. Keyboard navigation should be there. And styling is almost identical.

arcangelini committed:
Add accessibility

arcangelini committed:
Add accessibility label

arcangelini committed:
Fix esc button

jeherve reviewed (approved):
This is looking good for me. I noticed that while keyboard navigation between slides works, escaping to get out of the modal doesn't. I think it would be a nice improvement.

Edit: nevermind, you fixed it while I was testing. :)

Swiper as a dependency should be okay, we use the library in other products.

I think this should be good to go!

arcangelini committed:
Add transitions

arcangelini committed:
Add mobile efficiencies

arcangelini commented:
Looks like Swiper is causing some Jest to break. Working on a fix now.

arcangelini commented:
This fixes the test issues — https://github.com/Automattic/wp-calypso/pull/105555

Moved it to a separate PR to be safe.

arcangelini committed:
Ditch Swiper

arcangelini committed:
Discard changes to yarn.lock
